src/digraph-cyclicity.jl

```diff
@@ -0,0 +1,371 @@
+abstract Visitor
+
+"""
+```transitiveclosure!(dg::DiGraph)```
+
+Compute the transitive closure of a directed graph, using the Floyd-Warshall
+algorithm.
+"""
+function transitiveclosure!(dg::DiGraph)
+    for k in dg.vertices
+        for i in dg.vertices
+            for j in dg.vertices
+                if (has_edge(dg, i, k) & has_edge(dg, k, j))
```

Comment:
This is very expensive (`O(n^3)`). Could you iterate over the edges in `dg` instead? Something like...

```
for (i, k) in edges(dg)
  for j in neighbors(dg, k)
    add_edge!(dg, i, j)
```
(I haven't tested this but iterating over the known edges here will be more efficient than `has_edge` checks.)